frameworks/helloworld/tests/test_resource_refinement.py

```diff
@@ -26,10 +38,105 @@ def configure_package(configure_security):
         yield  # let the test session execute
     finally:
         sdk_install.uninstall(config.PACKAGE_NAME, config.SERVICE_NAME)
+        pass
 
 
 @pytest.mark.sanity
 @pytest.mark.smoke
 @pytest.mark.dcos_min_version('1.10')
 def test_install():
     config.check_running(config.SERVICE_NAME)
+
+
+@pytest.mark.sanity
+@pytest.mark.smoke
+@pytest.mark.dcos_min_version('1.10')
+def test_marathon_volume_collission():
+    # This test validates that a service registered in a sub-role of
+    # slave_public will _not_ unreserve Marathon volumes.
+
+    # Uninstall HW first
+    sdk_install.uninstall(config.PACKAGE_NAME, config.SERVICE_NAME)
+
+    # Install the marathon app
+    marathon_app_name = "persistent-test"
+    persistent_app = {
+        "id": marathon_app_name,
+        "mem": 128,
+        "user": "nobody",
+        "cmd": "echo 'this is a test' > persistent-volume/test && sleep 10000",
+        "container": {
+            "type": "MESOS",
+            "docker": {
+                "image": "ubuntu",
```

Comment:
Image needed at all? Could probably speed up the test if we drop this.

Reply:
Good point. Lemme see if you can do it without.